### PR TITLE
Add $size

### PR DESCRIPTION
**Pull request by johnsmartco**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by johnsmartco** [comment]
Ready for follow-up review @maheshrajamani.  Thanks!

File: docs/jsonapi-spec.md [hunk 1]
```diff
@@ -1419,26 +1419,26 @@ The `$all` operation allows users to check for documents that have an array fiel
...
 ```
 
-#### $elemMatch operation
+That expression returns, e.g.: `{ field: [ red, green ] }` and `{ field: [ apple, lime ] }`, but not `{ field: fruit }` or `{ field: [ orange, lemon, grapefruit ] }`.
 
-TODO:
+If the given field is not an array, there's no match. `$size` should ignore non-arrays.
 
-#### $size operation
+`$size` does not accept ranges of values. To select documents based on fields with different numbers of elements, a user could create a counter field to increment when adding elements to a field.
```
Comment (maheshrajamani) [inline docs/jsonapi-spec.md:1438]: These 3 lines are not required. The $size is based on pre build index. Also we don't support counter columns.

**Review by maheshrajamani — CHANGES_REQUESTED** [review]

File: docs/jsonapi-spec.md (hunk 1 above)
Comment (johnsmartco) [reply docs/jsonapi-spec.md:1438]: thanks will fix...

File: docs/jsonapi-spec.md [hunk 2]
```diff
@@ -1419,26 +1419,24 @@ The `$all` operation allows users to check for documents that have an array fiel
...
 ```
 
-#### $elemMatch operation
+That expression returns, e.g.: `{ field: [ red, green ] }` and `{ field: [ apple, lime ] }`, but not `{ field: fruit }` or `{ field: [ orange, lemon, grapefruit ] }`.
 
-TODO:
+If the given field is not an array, there's no match. `$size` should ignore non-arrays.
 
-#### $size operation
+Queries cannot use indexes for the $size portion of a query, although the other portions of a query can use indexes if applicable.
```
Comment (maheshrajamani) [inline docs/jsonapi-spec.md:1438]: Should have seen this before, $size index can be used along other filters and it uses the index. Can we remove this too?

**Review by maheshrajamani — APPROVED** [review]

File: docs/jsonapi-spec.md (hunk 2 above)
Comment (johnsmartco) [reply docs/jsonapi-spec.md:1438]: yes will do

File: docs/jsonapi-spec.md [hunk 3]
```diff
@@ -1419,26 +1419,22 @@ The `$all` operation allows users to check for documents that have an array fiel
...
 
-... is equivalent to:
+#### $size operation
 
-```json5
-{ $and: [ { tags: "ssl" }, { tags: "security" } ] }
+The `$size` operation allows users to match any array with the number of elements specified by the argument. Example:
 
+```json5
+db.collection.find( { field: { $size: 2 } } );
```
Comment (vkarpov15) [inline docs/jsonapi-spec.md:1431]: This isn't valid JSON. I think this should be `{ "field": { "$size": 2 } }`

File: docs/jsonapi-spec.md [hunk 4]
```diff
@@ -1419,26 +1419,22 @@ The `$all` operation allows users to check for documents that have an array fiel
...
 ```
 
-#### $elemMatch operation
-
-TODO:
+That expression returns, e.g.: `{ field: [ red, green ] }` and `{ field: [ apple, lime ] }`, but not `{ field: fruit }` or `{ field: [ orange, lemon, grapefruit ] }`.
```
Comment (vkarpov15) [inline docs/jsonapi-spec.md:1434]: Quotes around "field" and string values, so `{ "field": ["red", "green"] }`. We should use valid JSON where possible for consistency and so code samples are easier to copy/paste.

**Pull request merged**